controll/controller.js

```diff
@@ -32,12 +32,21 @@ function getArticleById(req, res, next){
 
 
 function getArticles(req, res, next){
-    fetchArticles().then((arrayOfArticles) => {
-        res.status(200).send({articles : arrayOfArticles})
-    })
-    .catch(next)
+    const { topic } = req.query
+    if (!topic){
+        fetchArticles().then((arrayOfArticles) => {
+            res.status(200).send({articles : arrayOfArticles})
+        })
+        .catch(next)
+    } else {
+        Promise.all([doesTopicExist(topic), fetchArticleByTopic(topic)]).then(([_, articles]) => {
```

Comment:
While this solution is fine, it would be possible to solve the topic query using you fetch articles model